### PR TITLE
glib: impl StaticType, FromValue, ToValue, HasParamSpec for Box<str>

### PR DESCRIPTION
**Pull request by SeaDve**
depends on https://github.com/gtk-rs/gtk-rs-core/pull/1043 (Of which commits are also included here)

### CODE REVIEW

File: glib-macros/tests/value_delegate_derive.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
-use glib::{value::FromValue, HasParamSpec, StaticType, ToValue};
+use glib::{value::FromValue, StaticType, ToValue};
 
 #[test]
 fn higher_level_types() {
     #[derive(Debug, glib::ValueDelegate)]
-    pub struct MyVec(Vec<String>);
```
Comment (sdroege) [inline glib-macros/tests/value_delegate_derive.rs]: Would be useful to have both tested or not?

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: glib-macros/tests/value_delegate_derive.rs (hunk 1 above)
Comment (SeaDve) [reply glib-macros/tests/value_delegate_derive.rs]: Both nullable and non, or both MyString and MyVec?

Comment (sdroege) [reply glib-macros/tests/value_delegate_derive.rs]: `MyString` and `MyVec`

Comment (ranfdev) [reply glib-macros/tests/value_delegate_derive.rs]: Yeah, let's keep both. They are good test cases for the ValueDelegate macro, which only had tests in the form of the example documentation.

File: glib/src/value.rs [hunk 2]
```diff
@@ -904,6 +904,49 @@ impl ToValueOptional for String {
...
+    }
+}
+
+impl ToValue for Box<str> {
+    fn to_value(&self) -> Value {
+        <&str>::to_value(&self.as_ref())
+    }
+
+    fn value_type(&self) -> Type {
+        String::static_type()
```
Comment (ranfdev) [inline glib/src/value.rs:933]: Why not self.value_type()?

Comment (SeaDve) [reply glib/src/value.rs:933]: I simply copied it from `String` implementation of `value_type`, and it seems that the implementation of `str` of `value_type` also calls into `String::static_type()`.

File: glib-macros/tests/value_delegate_derive.rs (hunk 1 above)
Comment (SeaDve) [reply glib-macros/tests/value_delegate_derive.rs:6]: Added both `MyVec` and `MyString`

**Review by sdroege — APPROVED** [review]

**Review by sdroege — APPROVED** [review]

**Pull request merged**